EventListener/RateLimitAnnotationListener.php

```diff
@@ -48,36 +33,39 @@ public function __construct(
         $this->pathLimitProcessor = $pathLimitProcessor;
     }
 
-    /**
-     * @param ControllerEvent|FilterControllerEvent $event
-     */
-    public function onKernelController($event)
+    public function onKernelController(ControllerEvent $event): void
     {
         // Skip if the bundle isn't enabled (for instance in test environment)
         if( ! $this->getParameter('enabled', true)) {
             return;
         }
 
         // Skip if we aren't the main request
-        if ($event->getRequestType() != HttpKernelInterface::MASTER_REQUEST) {
+        if ($event->getRequestType() !== HttpKernelInterface::MASTER_REQUEST) {
             return;
         }
 
-        // Find the best match
-        $annotations = $event->getRequest()->attributes->get('_x-rate-limit', array());
-        $rateLimit = $this->findBestMethodMatch($event->getRequest(), $annotations);
+        // RateLimits used to be set by sensio/framework-extra-bundle by reading annotations
+        // Tests also use that mechanism, we should probably keep it for retrocompatibility
+        if ($event->getRequest()->attributes->has('_x-rate-limit')) {
```

Comment:
nitpick: here you can extract `$event->getRequest()`  in a dedicated `$request` variable